src/iiptilesource.js

```diff
@@ -0,0 +1,319 @@
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. *
+ *
+ */
+
+
+(function($) {
+
+  /**
+   * @class IIPTileSource
+   * @classdesc A tilesource implementation for the Internet Imaging Protocol (IIP).
+   *
+   * @memberof OpenSeadragon
+   * @extends OpenSeadragon.TileSource
+   * @see https://iipimage.sourceforge.io
+   *
+   * @param {String} iipsrv               - IIPImage host server path (ex: "https://host/fcgi-bin/iipsrv.fcgi" or "/fcgi-bin/iipsrv.fcgi")
+   * @param {String} image                - Image path and name on server (ex: "image.tif")
+   * @param {String} format    (optional) - Tile output format (default: "jpg")
+   * @param {Object} transform (optional) - Object containing image processing transforms
+   *                                        (supported transform: "stack","quality","contrast","color","invert",
+   *                                                              "colormap," "gamma","minmax","twist","hillshade".
+   *                                        See https://iipimage.sourceforge.io/documentation/protocol for how to use)
+   *
+   * Example: tileSources: {
+   *            iipsrv:    "/fcgi-bin/iipsrv.fcgi",
+   *            image:     "test.tif",
+   *            transform: {
+   *              gamma: 1.5,
+   *              invert: true
+   *            }
+   *          }
+   */
+
+  $.IIPTileSource = function(options) {
+
+    $.EventSource.call( this );
+
+    if( options && options.image ){
+      $.extend( this, options );
+      this.aspectRatio = 1;
+      this.dimensions  = new $.Point( 10, 10 );
+      this._tileWidth  = 0;
+      this._tileHeight = 0;
+      this.tileOverlap = 0;
+      this.minLevel    = 0;
+      this.maxLevel    = 0;
+      this.ready       = false;
+
+      // Query server for image metadata
+      var url = this.iipsrv + '?FIF=' + this.image + '&obj=IIP,1.0&obj=Max-size&obj=Tile-size&obj=Resolution-number&obj=Resolutions';
+      this.getImageInfo( url );
+    }
+  };
+
+
+  $.extend($.IIPTileSource.prototype, $.TileSource.prototype, /** @lends OpenSeadragon.IIPTileSource.prototype */ {
+
+    /**
+     * Determine if the data and/or url imply the image service is supported by
+     * this tile source.
+     * @function
+     * @param {Object|Array} data
+     * @param {String} optional - url
+     */
+    supports: function(data, url) {
+      // Configuration must supply the IIP server endpoint and the image name
+      return ( data && ("iipsrv" in data) && ("image" in data) );
+    },
+
+
+    /**
+     * Parse IIP protocol response
+     * @function
+     * @param {Object|Array} data - raw metadata from an IIP server
+     */
+    parseIIP: function( data ) {
+
+      var tmp = data.split( "Max-size:" );
+      if(!tmp[1]){
+        throw new Error( "No Max-size returned" );
```

Comment:
Did these throws somehow get turned into `open-failed` events? I don't see the mechanism by which they would, but I'm rusty in that area. At any rate, I think that would be better than just throwing an exception.

Reply:
OK, all now switched to open-failed events